app/assets/javascripts/app/views/header_view.js

```diff
@@ -21,25 +27,118 @@ app.views.Header = app.views.Base.extend({
     new app.views.Notifications({ el: '#notification_dropdown' });
   },
 
-  menuElement : function() { return this.$("ul.dropdown"); },
+  badge: function(){ return $('#notification_badge'); },
+  dropdown: function(){ return $('#notification_dropdown'); },
+  dropdownNotifications: function(){ return this.dropdown().find(".notifications"); },
+  ajaxLoader: function(){ return this.dropdown().find(".ajax_loader"); },
+
+  toggleNotifDropdown: function(evt){
+    evt.preventDefault();
+    evt.stopPropagation();
+    if(this.notifDropdownShowing()){ this.hideNotifDropdown(evt); }
+    else{ this.showNotifDropdown(); }
+  },
+
+  notifDropdownShowing: function(){
+    return this.dropdown().css("display") === "block";
+  },
+
+  showNotifDropdown: function(){
+    this.ajaxLoader().show();
+    this.badge().addClass("active");
+    this.dropdown().css("display", "block");
+    this.dropdownNotifications().addClass("loading");
+    this.getNotifications();
+  },
+
+  hideNotifDropdown: function(evt){
+    var inDropdown = $(evt.target).parents().is(this.dropdown());
+    var inHovercard = $.contains(app.hovercard.el, evt.target);
+    if(!inDropdown && !inHovercard && this.notifDropdownShowing()){
+      this.badge().removeClass("active");
+      this.dropdown().css("display", "none");
+      this.dropdownNotifications().perfectScrollbar('destroy');
+    }
+  },
+
+  getMoreNotifications: function(page){
+    var self = this;
+    $.getJSON(Routes.notifications_path({ per_page:5, page: page }), function(notifications){
+      for(var i = 0; i < notifications.length; ++i)
+        self.notifications.push(notifications[i]);
```

Comment:
@AugierLe42e See https://github.com/diaspora/diaspora/blob/develop/config/.jshint.json#L4 and http://jshint.com/docs/options/#curly.